contracts/core/Dispatcher.sol

```diff
@@ -457,25 +480,27 @@ contract Dispatcher is IbcDispatcher, IbcEventsEmitter, Ownable, Ibc {
         consensusStateManager.verifyMembership(
             proof,
             packetCommitmentProofKey(packet),
-            abi.encode(packetCommitmentProofValue(packet))
+            bytes.concat(packetCommitmentProofValue(packet))
         );
 
         // verify packet has not been received yet
-        bool hasReceipt = recvPacketReceipt[address(receiver)][packet.dest.channelId][packet.sequence];
+        bool hasReceipt = packetCommitment[address(receiver)][packet.dest.channelId][packet.sequence].recvPacked;
         if (hasReceipt) {
             revert packetReceiptAlreadyExists();
         }
 
-        recvPacketReceipt[address(receiver)][packet.dest.channelId][packet.sequence] = true;
+        packetCommitment[address(receiver)][packet.dest.channelId][packet.sequence].recvPacked = true;
 
         // enforce recv'ed packet sequences always increment by 1 for ordered channels
         Channel memory channel = portChannelMap[address(receiver)][packet.dest.channelId];
         if (channel.ordering == ChannelOrder.ORDERED) {
-            if (packet.sequence != nextSequenceRecv[address(receiver)][packet.dest.channelId]) {
+            if (packet.sequence != nextSequence[address(receiver)][packet.dest.channelId].recv) {
                 revert unexpectedPacketSequence();
             }
 
-            nextSequenceRecv[address(receiver)][packet.dest.channelId] = packet.sequence + 1;
+            unchecked {
+                nextSequence[address(receiver)][packet.dest.channelId].recv = packet.sequence + 1;
```

Comment:
see other comment about potential overflow in the `acknowledgement` method

Reply:
I assmued that this check: `if (packet.sequence != nextSequence[address(receiver)][packet.dest.channelId].recv)` bounded it.